packages/gatsby/cache-dir/production-app.js

```diff
@@ -66,33 +67,35 @@ apiRunnerAsync(`onClientEntry`).then(() => {
       return (
         <EnsureResources location={location}>
           {({ pageResources, location }) => (
-            <RouteUpdates location={location}>
-              <ScrollContext
-                location={location}
-                shouldUpdateScroll={shouldUpdateScroll}
-              >
-                <Router
-                  basepath={__BASE_PATH__}
+            <StaticQueryContext.Provider value={pageResources.staticQueryData}>
```

Comment:
Does this work for queries in `wrapRootElement`? This provider is below those wrappers.

Do we have some test that would assert this working? If not we probably should add one

Reply:
Fixed that in [`32ceed6` (#25723)](https://github.com/gatsbyjs/gatsby/pull/25723/commits/32ceed6fd3f85017b1956f9d1d7f7d0e03f5481b)

Note that while we did have a test to assert this (https://github.com/gatsbyjs/gatsby/blob/c82569f68ac233360e1d8d4db428c69149da565a/e2e-tests/production-runtime/cypress/integration/static-query.js#L22), the test passing seems to have been a false positive? The text it was looking for would be in DOM because of SSR even if static query data was not loaded in JS (which would fail hydration but the test would _still_ pass because the SSRed content is _still_ correct)

Reply:
@pieh I've commented out the `gatsby-ssr` for the moment to _actually_ assert this.